src/view/range.js

```diff
@@ -82,6 +87,60 @@ export default class Range {
 	}
 
 	/**
+	 * Creates a new range with the same content, but containing all surrendering tags with no semantic meaning.
+	 *
+	 * For example:
+	 *
+	 * 		<p>Foo</p><p><b>{Bar}</b></p> -> <p>Foo</p>[<p><b>Bar</b></p>]
+	 * 		<p><b>foo</b>{bar}<span></span></p> -> <p><b>foo[</b>bar<span></span>]</p>
+	 *
+	 * Note that in the sample above all:
+	 *  - `<p>` have type of {@link module:engine/view/containerelement~ContainerElement},
+	 *  - `<b>` have type of {@link module:engine/view/attributeelement~AttributeElement},
+	 *  - `<span>` have type of {@link module:engine/view/uielement~UiElement}.
+	 *
+	 * @returns {module:engine/view/range~Range} Enlarged range.
+	 */
+	getEnlarged() {
+		const start = this.start.getPriorPosition( enlagreShrinkStartSkip );
+		const end = this.end.getFurtherPosition( enlagreShrinkEndSkip );
+
+		return new Range( start, end );
+	}
+
+	/**
+	 * Creates a new range with the same content, but without all surrendering tags with no semantic meaning.
+	 *
+	 * For example:
+	 *
+	 * 		<p>Foo</p>[<p><b>Bar</b></p>] -> <p>Foo</p><p><b>{Bar}</b></p>
+	 * 		<p><b>foo[</b>bar<span></span>]</p> -> <p><b>foo</b>{bar}<span></span></p>
+	 *
+	 * Note that in the sample above all:
+	 *  - `<p>` have type of {@link module:engine/view/containerelement~ContainerElement},
+	 *  - `<b>` have type of {@link module:engine/view/attributeelement~AttributeElement},
+	 *  - `<span>` have type of {@link module:engine/view/uielement~UiElement}.
+	 *
+	 * @returns {module:engine/view/range~Range} Shrink range.
+	 */
+	getShrinked() {
+		let start = this.start.getFurtherPosition( enlagreShrinkStartSkip );
+		let end = this.end.getPriorPosition( enlagreShrinkEndSkip );
+		let nodeAfterStart = start.nodeAfter;
+		let nodeBeforeEnd = end.nodeBefore;
+
+		if ( nodeAfterStart instanceof Text ) {
```

Comment:
I'll add a comment that "Technically, position at the beginning of a text node is after position before that node, so we are fixing this because TreeWalker does not enter text nodes"